optaplanner-examples/src/main/java/org/optaplanner/examples/nqueens/domain/solver/nearby/QueenNearbyDistance.java

```diff
@@ -0,0 +1,16 @@
+package org.optaplanner.examples.nqueens.domain.solver.nearby;
```

Comment:
Personally, I wouldn't touch nqueens at all.
The example is so trivial that introducing ruin doesn't help it much, but it does introduce extra complexity, which makes the "teaching value" of the example smaller.